src/jitlog/reader.lua

```diff
@@ -303,6 +303,9 @@ function base_actions:protobl(msg)
     bcindex = msg:get_bcindex(),
     time = msg.time,
   }
+  if not proto.blacklisted then
+    proto.blacklisted = blacklist
+  end
```

Comment:
Shouldn't this be an `else`? I'm not sure if line 309 makes sense if line 307 has been executed?

Reply:
Its just storing the first time the function proto got black listed in to the function proto this event was for. I did start out with support for it making a linked list but it didn't really make sense.

Reply:
I'm not sure that really answers my question. Isn't it storing the same thing twice?

Reply:
'blacklist' is the blacklist event object not a list, 'proto_blacklist' is a list of blacklist events and proto is function that this event is for

Reply:
Ah.